feat/tests/test_fast_detector.py

```diff
@@ -35,3 +44,225 @@ def test_detect_image(self, single_face_img):
 
         # AU checks; TODO: Add more
         assert fex.aus.AU20[0] > 0.8
+
+        
+    def test_fast_landmark_with_batches(self, multiple_images_for_batch_testing):
+        """
+        Make sure that when the same images are passed in with and without batch
+        processing, the detected landmarks and poses come out to be the same
+        """
+        det_result_batch = self.detector.detect(
+            inputs=multiple_images_for_batch_testing,
+            batch_size=5,
+        )
+
+        det_result_no_batch = self.detector.detect(
+            inputs=multiple_images_for_batch_testing,
+            batch_size=1,
+        )
+
+        assert np.allclose(
+            det_result_batch.loc[:, "x_0":"y_67"].to_numpy(),
+            det_result_no_batch.loc[:, "x_0":"y_67"].to_numpy(),
+        )
+            
+    # TODO: Currently making this test always pass even if batching gives slightly diff
+    # results until @tiankang can debug whether we're in tolerance
+    # Track progress updates in this issue: https://github.com/cosanlab/py-feat/issues/128
+    def test_fast_detection_and_batching_with_diff_img_sizes(
+        self, single_face_img, multi_face_img, multiple_images_for_batch_testing
+    ):
+        """
+        Make sure that when the same images are passed in with and without batch
+        processing, the detected landmarks and poses come out to be the same
+        """
+        # Each sublist of images contains different sizes
+        all_images = (
+            [single_face_img] + [multi_face_img] + multiple_images_for_batch_testing
+        )
+
+        # Multiple images with different sizes are ok as long as batch_size == 1
+        # Detections will be done in each image's native resolution
+        det_output_default = self.detector.detect(inputs=all_images, batch_size=1)
+
+        # TODO: fast_detector does not currently raise ValueError
+        # If batch_size > 1 then output_size must be set otherwise we can't stack to make a
+        # tensor
+        # with pytest.raises(ValueError):
+        #     _ = self.detector.detect(
+        #         inputs=all_images,
+        #         batch_size=5,
+        #     )
+
+        # Here we batch by resizing each image to 256xpadding
+        batched = self.detector.detect(
+            inputs=all_images, batch_size=5, output_size=256
+        )
+
+        # We can also forcibly resize images even if we don't batch process them
+        nonbatched = self.detector.detect(
+            inputs=all_images, batch_size=1, output_size=256
+        )
+
+        # To make sure that resizing doesn't interact unexpectedly with batching, we should
+        # check that the detections we get back for the same sized images are the same when
+        # processed as a batch or serially. We check each column separately
```

Comment:
Can you add a note here that it looks like AU25 is the only one showing non-zero deviance between batched and non-batched processing? Actually it would be good to verify if that's true on your machine as well. Something like this should work and you can just see what AU's have values > 0: `(batched.aus - nonbatched.aus).abs().sum(0)`

Reply:
Looks like my machine is outputting all values of 0.0. I used the above statement and ran the test a few times to be sure, and the max AU deviation printed as 0.0 for AU1 every time.